src/ducks/auth.ts

```diff
@@ -1701,5 +1727,58 @@ export const useAuthenticationStore = create<AuthStore>()((set, get) => {
       password: string,
       activeAccountId?: string | null,
     ) => getKeyFromKeyManager(password, activeAccountId),
+
+    importSecretKey: async (secretKey: string, password: string) => {
+      set({ isLoading: true, error: null });
+
+      try {
+        const keypair = Keypair.fromSecret(secretKey);
+        const publicKey = keypair.publicKey();
+        const privateKey = keypair.secret();
+
+        const existingAccounts = await getAllAccounts();
+        const accountExists = existingAccounts.some(
+          (account) => account.publicKey === publicKey,
+        );
+
+        if (accountExists) {
+          throw new Error(t("authStore.error.accountAlreadyExists"));
+        }
+
+        const temporaryStore = await getTemporaryStore();
+        if (!temporaryStore) {
+          throw new Error(t("authStore.error.temporaryStoreNotFound"));
+        }
+
+        const keyPair = {
+          publicKey,
+          privateKey,
+        };
+
+        await storeAccount({
+          mnemonicPhrase: temporaryStore.mnemonicPhrase,
+          password,
+          keyPair,
+          accountNumber: existingAccounts.length + 1,
+          shouldRefreshHashKey: false,
+          shouldSetActiveAccount: true,
+          imported: true,
+        });
```

Comment:
Could we extract this chunk of code to a local `importSecretKey` function so we keep this auth action clean?
The same we are doing with `createAccount`, something like:

```
    importSecretKey: async (...) => {
      set({ ... });

      try {
        await importSecretKey(...);

        await Promise.all([get().getAllAccounts(), get().fetchActiveAccount()]);

        set({ ... });
      } catch (error) {
        set({ ... });
        throw error;
      }
    },
```